specification/attributes/discount_handling.md

```diff
@@ -0,0 +1,49 @@
+
+While providers may use different terms to describe discounts, FOCUS identifies a discount as being a reduced price applied directly to a row. Any price or cost reductions that are awarded after the fact are identified as a "Credit" Charge Subcategory. One example might be when a provider offers a reduced rate after passing a certain threshold of usage or spend.
+
+All rows defined in FOCUS MUST follow the discount handling requirements listed below.
+
+## Attribute ID
+
+DiscountHandling
+
+## Attribute name
+
+Discount Handling
+
+## Description
+
+Indicates how to include and apply discounts to usage charges or rows.
+
+## Requirements
+
+* All applicable discounts SHOULD be applied to each row they pertain to and SHOULD NOT be negated in a separate row.
+* All discounts applied to a row MUST apply to the entire charge.
+  * If a discount only applies to a subset of the charge, then that portion of the charge MUST be split into a separate row. One with the discount and one without.
+  * If multiple discounts apply, there MUST be separate rows for each unique combination of applied discounts or undiscounted amount.
+  * Each discount MUST be identifiable using existing FOCUS columns.
+    * Discounts applied to rows from a commitment-based discount MUST include a CommitmentDiscountId.
+    * If a provider applies a discount that cannot be represented by a FOCUS column, they SHOULD include additional columns to identify the source of the discount.
+* Purchased discounts (e.g., commitment-based discounts) MUST specify CommitmentUtilization on the usage rows that received the reduced rates from the discount.
+* Purchased discounts (e.g., commitment-based discounts) MUST be amortized.
+  * The BilledCost MUST be 0 for any row where the commitment covers the entire cost for the charge period.
+  * The EffectiveCost MUST be the portion of the amortized purchase cost that applies to this row.
+  * CommitmentUtilization MUST be "Used" for rows that received a reduced price from that commitment.
+  * If a commitment is not fully utilized, the provider MUST include a row that represents the unused portion of the commitment for that charge period. Charge Subcategory MUST be "Commitment Not Used".
+  * The sum of the EffectiveCost for all "Commitment Used" and "Commitment Not Used" rows for each ChargeSubcategory MUST be the same as the BilledCost of the commitment-based discount purchase.
```

Comment:
This won't work. I see a few issues:
- First, I think you mean to say the sum of Used and Unused for the duration of the commitment should equal the billed cost of the purchase.
- The above statement would be true for all up-front paid RIs. In the case of partial and no up-front RIs, the effective cost would be the amortized portion + on-going rate you pay? So the sum will not equal the purchase
- In cases like partial or no up-front, are we suggesting there is a need for a single row that represents the 'purchase' record in each lowest-granularity charge period?

```suggestion
  * The sum of the EffectiveCost for all "Used Commitment" and "Unused Commitment" rows for each ChargeSubcategory MUST be the same as the BilledCost of the commitment-based discount purchase.
```